appservice/src/intTest/groovy/nu/yona/server/OverwriteUserTest.groovy

```diff
@@ -1,220 +1,184 @@
 package nu.yona.server
 
 import groovy.json.*
-import spock.lang.Shared
-
-class OverwriteUserTest extends AbstractAppServiceIntegrationTest {
-
-	@Shared
-	def richardQuin
-	@Shared
-	def bobDunn
-
-	@Shared
-	def bobDunnRichardBuddyURL
-
-	@Shared
-	def richardQuinNewURL
-	@Shared
-	def richardQuinNewPassword = "R i c h a r d 1"
-	@Shared
-	def richardQuinOverwriteConfirmationCode
-
-	@Shared
-	def userCreationMobileNumber = "+${timestamp}"
-	@Shared
-	def userCreationJSON = """{
-				"firstName":"John",
-				"lastName":"Doe ${timestamp}",
-				"nickname":"JD ${timestamp}",
-				"mobileNumber":"${userCreationMobileNumber}",
-				"devices":[
-					"Galaxy mini"
-				],
-				"goals":[
-					"gambling"
-				]}"""
-
-	def 'Add Richard and Bob who are buddies'(){
-		given:
-
-		when:
-		richardQuin = addUser("Richard", "Quin")
-		bobDunn = addUser("Bob", "Dunn")
-		makeBuddies(richardQuin, bobDunn)
 
-		then:
-		richardQuin
-		bobDunn
-	}
+import nu.yona.server.YonaServer
 
-	def 'Classification engine detects a potential conflict for Richard'(){
+class OverwriteUserTest extends AbstractAppServiceIntegrationTest
+{
+	def 'Attempt to add another user with the same mobile number'()
+	{
 		given:
+			def richard = addRichard()
 
 		when:
-		def response = analysisService.postToAnalysisEngine("""{
-				"vpnLoginID":"${richardQuin.vpnLoginID}",
-				"categories": ["news/media"],
-				"url":"http://www.refdag.nl"
-				}""")
+			def duplicateUser = appService.addUser(this.&userExistsAsserter, "A n o t h e r", "The", "Next", "TN",
+				"$richard.mobileNumber", [ "Nexus 6" ], [ "news", "gambling" ])
 
 		then:
-		response.status == 200
-	}
-
-	def 'Attempt to add another user with the same mobile number'(){
-		when:
-		def response = appService.addUser("""{
-					"firstName":"Richardo ${timestamp}",
-					"lastName":"Quino ${timestamp}",
-					"nickname":"RQo ${timestamp}",
-					"mobileNumber":"${richardQuin.mobileNumber}",
-					"devices":[
-						"Nexus 6"
-					],
-					"goals":[
-						"news"
-					]
-				}""", "Foo")
+			duplicateUser == null
 
-		then:
-		response.status == 400
-		response.responseData.code == "error.user.exists"
+		cleanup:
+			appService.deleteUser(richard)
 	}
 
-	def 'Request overwrite of the existing user'(){
-		when:
-		def response = appService.requestOverwriteUser("${richardQuin.mobileNumber}")
-		if(response.status == 200) {
-			richardQuinOverwriteConfirmationCode = response.responseData.confirmationCode
-		}
-		then:
-		response.status == 200
-		richardQuinOverwriteConfirmationCode
+	def userExistsAsserter(def response)
```

Comment:
The test "Attempt to add another user with the same mobile number" passes it as asserter to the ``addUser`` call.